### ISSUE TITLE
How to change the language?

### ISSUE THREAD

**Issue by MrFredber** [zombieFox/nightTab#329]
I couldn't find a setting, that can change the language. If the language depends on the chrome language, then why this isn't working?

![image](https://user-images.githubusercontent.com/33953266/146644402-1536e08f-83a9-40c2-8e3f-2c86711ed026.png)

Desktop:
 - OS: Windows 10 21H2
 - Browser: Chrome
 - Version: 96.0.4664.110
 - nightTab version: 7.3.0

**Comment by zombieFox** [comment]
Hi, sorry but the app only supports English for now. Language support ([WIP](https://github.com/zombieFox/nightTab/tree/lang)) is something I've been meaning to look into for a while.

**Comment by OckhamOdyssey** [comment]
I get a 404 error while loading the link. Anyway, if you need a Spanish translator, I would be pleased to help.

**Comment by metruzanca** [comment]
That link was to a branch that where the language support was being worked on that is now merged into main it seems. (note, that was about 5 months ago) Language support is almost done. Its live on the web version but not quite in the browser extension. Not sure whats preventing us from getting it in the extension. @zombieFox would probably be able to provide insight on this.

**Comment by MrFredber** [comment]
That's glad to hear, but web version? I never heard this before. Where i can see this version?

**Comment by metruzanca** [comment]
@MrFredber if you go to the main page on the github repo, you should see the link in the upper right as shown here
![image](https://user-images.githubusercontent.com/32832892/170639896-28a83035-842c-4d93-bf2f-192588f0fd6e.png)

That spot on github is reserved for URLs relating to the project, so you'll usually find the webapp url there. If its not a webapp, you'll usually get the website relating to the project.

The web version is basically the demo/beta to any new features in development. This is because its super easy to deploy a new version for the web, meanwhile getting new versions for the extensions is a bit more time consuming and this way we can verify everything works from multiple devices easily.

(_on a side note, ZombieFox handles extension releases. I'm just giving him a hand with issue replies 😄_)

**Comment by metruzanca** [comment]
For those curious, once the language feature is fully added to the extensions, it'll look something like this: 
> ![image](https://user-images.githubusercontent.com/32832892/170640661-921bf954-afd4-4e73-8404-0eab67ca7de4.png)
> _Screenshot from the [web version](https://zombiefox.github.io/nightTab/)_

(The transparent menu is due to the fact that when I take a screenshot the menu starts disappearing)

---

That being said, I don't have a timeline for you guys on when that'll be out.

Also no clue how ZombieFox got the translations, (maybe they're google translated? 🤷🏻) so we'd be more than happy if you wanted to help us improve the spanish version @OckhamOdyssey. I myself will probably take a look at the italian version at some point.

If you want to get your hands dirty already, you can either use the web version then cross refrencing this file https://github.com/zombieFox/nightTab/blob/main/src/locale/es/messages.json or using the file directly. Then if you want to contribute, feel free to make a PR.

**Comment by OckhamOdyssey** [comment]
Yes, I have seen the Language option on the web version. Doesn't seem natively translated, at least in Spanish. Also, the greetings and the day of the week is not translated.

**Comment by zombieFox** [comment]
The bulk of the language support is not merged in to main. This means following the [Chrome i18n](https://developer.chrome.com/docs/extensions/reference/i18n/) spec. 

The language select works on nightTab, text strings are pulled from the [appropriate locale message JSON](https://github.com/zombieFox/nightTab/tree/main/src/locale). However after merging I discovered the build process bundle two copies of the message JSON files into the extension. This problem is still being investigated. Until this is resolved I can't update the browser extensions.

**Comment by zombieFox** [comment]
> Also, the greetings and the day of the week is not translated.

These header elements are not likely to be translated as the syntax for different languages plus the customisation of the header strings is too complex to support.
